gitlabbot/gitlabbot/util.go

```diff
@@ -57,16 +58,30 @@ func formatPipelineMsg(evt *gitlab.PipelineEvent, username string) (res string)
 	}
 }
 
-func formatSetupInstructions(repo string, msg chat1.MsgSummary, httpAddress string, secret string) (res string) {
+func formatSetupInstructions(repo string, hostedURL string, msg chat1.MsgSummary, httpAddress string, secret string) (res string) {
 	back := "`"
 	message := fmt.Sprintf(`
-To configure your project to send notifications, go to https://gitlab.com/%s/-/settings/integrations and add a new webhook.
+To configure your project to send notifications, go to %s/%s/-/settings/integrations and add a new webhook.
 For “URL”, enter %s%s/gitlabbot/webhook%s.
 For “Secret Token”, enter %s%s%s.
 Remember to check all the triggers you would like me to update you on.
 Note that I currently support the following Webhook Events: Push, Issues, Merge Request, Pipeline
 
 Happy coding!`,
-		repo, back, httpAddress, back, back, base.MakeSecret(repo, msg.ConvID, secret), back)
+		hostedURL, repo, back, httpAddress, back, back, base.MakeSecret(repo, msg.ConvID, secret), back)
 	return message
 }
+
+// parseRepoInput checks if url or <owner/repo> form
+func parseRepoInput(urlOrRepoPath string) (hostedURL string, repo string) {
+	parsedURL, err := url.ParseRequestURI(urlOrRepoPath)
+	if err != nil {
+		repo = urlOrRepoPath
+		hostedURL = "https://gitlab.com"
+	} else {
+		hostedURL = parsedURL.Scheme + "://" + parsedURL.Host
```

Comment:
we should check if the scheme/host are set. right now `/owner/repo` get's misparsed as a valid URL. can you add some unit tests for parseRepoInput for some common cases?